pkg/cfn/manager/api.go

```diff
@@ -552,8 +552,34 @@ func defaultStackStatusFilter() []types.StackStatus {
 	return allNonDeletedStackStatuses()
 }
 
-// DeleteStackBySpec sends a request to delete the stack
-func (c *StackCollection) DeleteStackBySpec(ctx context.Context, s *Stack) (*Stack, error) {
+// DeleteStack sends a request to delete the stack.
+// If Wait is true, it waits until status is DELETE_COMPLETE;
+// If ErrCh is given, any errors will be written to it, assume completion when nil is written,
+// do not expect more than one error value on the channel, it's closed immediately after it is written to.
+// Wait and ErrCh are mutually exclusive
+func (c *StackCollection) DeleteStack(ctx context.Context, options DeleteStackOptions) error {
+	if _, err := c.doDeleteStack(ctx, options.Stack); err != nil {
```

Comment:
I may be missing something, but I don't see much value in this function returning back the stack? Could we perhaps refactor here as-well such that it only returns an error?

Reply:
As `doDeleteStack()` is only used here, there is indeed no point in returnning the stack.
I made the change on `doDeleteStack()`.